### PR TITLE
Run tests with new kind setup script

### PR DESCRIPTION
**Pull request by su225**
~~**DO NOT MERGE** - This PR is for testing new KinD script to make sure that things don't break. There can be even changes in `common` directory purely for experimentation purposes. When it is finalized, it would be merged back to `common-files`, rough edges will be removed and we would be good to merge.~~

This is part of multicluster test infrastructure setup and deduplicating KinD cluster setup in istio/istio and istio/istio.io repos. (Bonus - you also get to define arbitrary cluster topologies)
https://github.com/istio/common-files/pull/311 

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[X] Developer Infrastructure

### CODE REVIEW

**Comment by su225** [comment]
@frankbu - There is plenty of work left here.

**Review by nmittler — APPROVED** [review]

**Comment by nmittler** [comment]
@su225 so just to be clear ... this will spin up 5 cluster by default.  That will slow down the tests however, so we'll probably want to introduce a second config for single cluster and have the prow jobs pass in the topology explicitly.

**Comment by su225** [comment]
@nmittler - Oops. My bad. I'll fix config.

**Comment by su225** [comment]
@nmittler - No. That is a single cluster. (https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/8127/doc.test.profile_default_istio.io/43/build-log.txt). The multicluster config has 3 clusters (https://github.com/istio/istio.io/pull/8127/files#diff-a89c5daba90954f65ca3661b06bc2774)

File: Makefile.overrides.mk [hunk 1]
```diff
@@ -19,4 +19,5 @@ CONTAINER_OPTIONS = -p 1313:1313 ${ADDITIONAL_CONTAINER_OPTIONS}
 BUILD_WITH_CONTAINER ?= 1
 
 # The release branching scripts use git commands so mount the global .gitconfig into the container
-CONDITIONAL_HOST_MOUNTS = --mount type=bind,source=${HOME}/.gitconfig,destination=/config/.gitconfig,readonly ${}
\ No newline at end of file
+CONDITIONAL_HOST_MOUNTS = --mount type=bind,source=${HOME}/.gitconfig,destination=/config/.gitconfig,readonly \
+													--mount type=bind,source=/tmp,destination=/tmp ${}
```
Comment (frankbu) [inline Makefile.overrides.mk:23]: Small nit:
```suggestion
CONDITIONAL_HOST_MOUNTS = --mount type=bind,source=${HOME}/.gitconfig,destination=/config/.gitconfig,readonly \
                          --mount type=bind,source=/tmp,destination=/tmp ${}
```

**Review by frankbu — APPROVED** [review]
LGTM

**Pull request merged**